### DISCUSSION TITLE
Where are saved searches?

### DISCUSSION THREAD

**Discussion by LorenAmelang** [TriliumNext/Trilium#2922] [Q&A]
My Ctrl+S searches are being saved somehow. If I search for "search" I get hundreds of "Search: ... (hidden / search)" results. 

But (unlike the previous discussion here) they are not in my Journal, and do not appear in my new Note with label #searchHome. 

Maybe they are hidden? But where? How can I show them? 

If I show the list in Full Search, they all say "not executed yet", even if they were in the past - is that the intention? 

There are often multiple copies of the same search, with identical appearance - is there a benefit to that? 

Here's a result that looks like it is from just now:
[searches](http://127.0.0.1:37840/#MptZutI25qLA) (Loren)#searchHome
I just made the "searches" Note with Owned Attribute #searchHome ...
Oh, wait, that's not a saved search like the previous ones, it is a regular note, like all the following lines! Doesn't have "#searchString=" included. 
Somehow hundreds of regular notes show up as search results...  Guess they do have "search" or "research" in them somewhere. 

Is there a way to see just the recently saved search results?

**Comment by zadam** [comment]
Hi, you're right in that searches are saved. Trilium has the concept of "saved search" where the user wants to keep a search, and it's represented as a note. Because saved search and regular search share the implementation, it's easier when even the regular searches operate as a note. The difference here is that "regular search" should be hidden, which is not the case here, and it's "leaking".

Implementation wise, root note contains a hidden subtree which contains these hidden notes.

`#searchHome` is something different, it marks a default location for saved search notes.

> Is there a way to see just the recently saved search results?

No, there isn't, but yeah, this is one of the possibilities where the current concept of storing searches could lead.

**Reply by LorenAmelang** [reply]
I see now how using the "Save Search" button in Full Search works, and those do go into the #searchHome note. 

I guess all the unsaved searches use an insignificant amount of data space, not worth worrying about? But they do clutter the Quick Search results list with multiple copies of the same search...  Guess I should save searches I might want again, to avoid the duplicates. 

Is there any way to see the "hidden subtree" and delete them?

**Reply by zadam** [reply]
In https://github.com/zadam/trilium/commit/0e01c19414fa15107304ba924d03dafd16dceb02 I fixed these searches showing up in quick search so that shouldn't be a problem.

I plan to provide a way to manage this search history, including cleaning them up. In the meantime, I wouldn't worry about it, the cost of an individual search note is very small.

**Reply by DHS1001** [reply]
> I plan to provide a way to manage this search history, including cleaning them up. In the meantime, I wouldn't worry about it, the cost of an individual search note is very small.

+1 
It would be great to have the functions to manage the search history.

**Comment by LorenAmelang** [comment]
Not sure how I triggered it, maybe by a QuickSearch, but a "Hidden Notes" subtree appeared at the bottom of my root tree under (and equal level to) "Components". I opened it and saw maybe a dozen interesting looking branches. I clicked the top one, which was a Note Map (that seemed to show all of my notes), and the Hidden subtree disappeared again. 

I see https://github.com/zadam/trilium/issues/3519: When switching "Hide archived notes", the hidden note subtree is visible. 

Found Menu -> Advanced -> Show Hidden Subtree - selecting that shows _only_ the Hidden tree! 

But selecting it again doesn't un-show it...  Turns out you have to close all the items you opened from it, and then it disappears. 

It has a Search History branch, with one item, that is not in my saved searches folder, and is way older than my latest unsaved search. No idea why that one...  That opens to show branches of all the found notes. 

No big deal, but my curiosity is certainly up...

**Reply by zadam** [reply]
> I see https://github.com/zadam/trilium/issues/3519: When switching "Hide archived notes", the hidden note subtree is visible.

Do you use the latest stable?

> But selecting it again doesn't un-show it... Turns out you have to close all the items you opened from it, and then it disappears.

Not sure what you mean TBH. You can either close the tab or unhoist from the hidden subtree:

![image](https://user-images.githubusercontent.com/617641/218279459-0aeee6e1-bb25-4075-a1b3-812a3cc740be.png)

> It has a Search History branch, with one item, that is not in my saved searches folder, and is way older than my latest unsaved search. 

By "my save searches folder" do you mean you have #searchHome defined?

> and is way older than my latest unsaved search.

Could you please make a screenshot of how your tree (only the structure) of hidden subtree looks like (with search history expanded)?

**Reply by LorenAmelang** [reply]
@zadam I'm using 0.58.7 on Windows for this. 

` You can either close the tab or unhoist from the hidden subtree:`

Thanks! Now I get what "unhoist" means...  

`By "my save searches folder" do you mean you have #searchHome defined?`

![image](https://user-images.githubusercontent.com/1297090/218283192-5aeb692a-47f0-4948-ba45-fef4bf9d6b9f.png)

![image](https://user-images.githubusercontent.com/1297090/218282784-c76df638-5e0e-484a-ac20-985fc5a37568.png)

`Could you please make a screenshot of how your tree (only the structure) of hidden subtree looks like (with search history expanded)?`

![image](https://user-images.githubusercontent.com/1297090/218282907-c2edc9c2-2d2f-47db-b0a2-5d5dd6d0a6b5.png)

Lots of unsaved searches since that one...  Aha! if I mouse over that search item and click the spin arrows icon, I get a popup - "Refresh saved search results" - so I guess it thinks that was saved? Got the same 25 hits as before. 

Appreciate the clues! But still unsure how this is supposed to work...

**Reply by LorenAmelang** [reply]
Aha! Only Quick Searches that you "Show in full search" appear in the Hidden note list. And those only appear in #searchHome Saved Searches if you explicitly save them. That makes sense, once you know it...  

I guess I was confused by the ancient issue of hundreds of "Search: ... (hidden / search)" results that began this thread. Seemed every quick search showed then.

**Reply by zadam** [reply]
Okay, so that all sounds like it's working as expected.

* searchHome is only for the searches you explicitly want to have as saved
* the hidden "Search History" are "full" searches which are not explicitly saved into the main (non-hidden) tree.

**Reply by LorenAmelang** [reply]
Still 0.58.7...  Last night I manually closed all my notes and quit Trilium. Did a hard restart of Windows 10 - hold power button for 20+ seconds, then wait, and power on. (Explorer gets stuck in a wait chain if I do too much interaction with iCloud...) Put it to sleep, woke it just now. When I clicked the pinned Trilium icon in the taskbar, I got this: 

![2023-02-12](https://user-images.githubusercontent.com/1297090/218336493-53812416-3b88-4de9-af93-3176938416d6.png)

All live and working, I explored a bit, then unhoisted it and got my normal view. Quit Trilium, started it again, didn't repeat. 

But I did figure out how I got the "Hidden Notes" subtree at the bottom of my root tree. Any time I click the "Collapse note tree" button along the bottom of the pane, it adds the Hidden Notes item! With no hoist icon. 100% repeatable. Open any subnote and it disappears. 

Does this tell you anything?
```
3:07:25.295 Sending message to all clients: {"type":"sync-finished","lastSyncedPush":179320}
13:07:25.523 Slow recursive query took 25ms.
13:07:25.554 Slow 200 PUT /api/branches/none_root/expanded-subtree/0 with 133773 bytes took 56ms
13:07:25.563 200 GET /api/tree?subTreeNoteId=root with 6899 bytes took 1ms
13:07:34.678 200 GET /api/tree?subTreeNoteId=Bk5u4SH6jD5R with 4169 bytes took 1ms
13:07:34.895 204 PUT /api/branches/root_Bk5u4SH6jD5R/expanded/1 with 0 bytes took 0ms
13:07:43.190 200 GET /api/notes/wJilMcBKMoSv with 2556 bytes took 0ms
13:07:43.191 200 GET /api/note-map/wJilMcBKMoSv/backlink-count with 11 bytes took 1ms
13:07:43.212 204 PUT /api/options with 0 bytes took 2ms
13:07:49.365 Slow 200 GET /api/search/%23workspace%20%23!template with 2 bytes took 20ms
13:07:49.387 204 PUT /api/options with 0 bytes took 2ms
13:07:58.094 304 GET /api/clipper/handshake with 45 bytes took 1ms
13:07:58.674 Slow recursive query took 21ms.
13:07:58.705 Slow 200 PUT /api/branches/none_root/expanded-subtree/0 with 133773 bytes took 52ms
13:07:58.710 200 GET /api/tree?subTreeNoteId=root with 6899 bytes took 0ms
13:08:02.435 200 GET /api/tree?subTreeNoteId=_hidden with 8232 bytes took 2ms
13:08:02.653 204 PUT /api/branches/root__hidden/expanded/1 with 0 bytes took 1ms
13:08:11.048 200 GET /api/note-map/_backendLog/backlink-count with 11 bytes took 1ms

```

**Reply by LorenAmelang** [reply]
Maybe more useful - the first startup of the day, out to my repeatability test: 

```
12:36:18.535 DB size: 274251 KB
12:36:20.557 Trusted reverse proxy: false
12:36:20.557 App HTTP server starting up at port 37840
12:36:20.557 {
  "appVersion": "0.58.7",
  "dbVersion": 212,
  "syncVersion": 29,
  "buildDate": "2023-01-17T23:14:58+01:00",
  "buildRevision": "a3149aecf41bac3c559ebbd1865e916264985ac3",
  "dataDirectory": "C:\\Users\\loren\\AppData\\Roaming\\trilium-data",
  "clipperProtocolVersion": "1.0",
  "utcDateTime": "2023-02-12T20:36:18.533Z"
}
12:36:20.558 CPU model: Intel(R) Core(TM) i7-6600U CPU @ 2.60GHz, logical cores: 4 freq: 2808 Mhz
12:36:20.576 Listening on port 37840
12:36:20.657 Slow query took 79ms: SELECT noteId, title, type, mime, isProtected, dateCreated, dateModified, utcDateCreated, utcDateModified FROM notes WHERE isDeleted = 0
12:36:20.711 Slow query took 43ms: SELECT branchId, noteId, parentNoteId, prefix, notePosition, isExpanded, utcDateModified FROM branches WHERE isDeleted = 0
12:36:20.842 Slow query took 105ms: SELECT attributeId, noteId, type, name, value, isInheritable, position, utcDateModified FROM attributes WHERE isDeleted = 0
12:36:20.886 Becca (note cache) load took 309ms
12:36:21.189 Keyboard action showNoteInfo found in database, but not in action definition.
12:36:21.189 Keyboard action showLinkMap found in database, but not in action definition.
12:36:21.189 Keyboard action toggleZenMode found in database, but not in action definition.
12:36:21.189 Keyboard action focusOnAttributes found in database, but not in action definition.
12:36:21.190 Registered global shortcut Ctrl+Alt+P for action createNoteIntoInbox
12:36:21.667 Generated CSRF token i7leNDHp-G5wC2rFPbbUHdlv2my4iFIv7LyM with secret _csrf=mI1ChdowjaIHFA2KP647qTRL; Path=/
12:36:22.219 200 GET /api/options with 6600 bytes took 1ms
12:36:22.221 200 GET /api/tree with 6898 bytes took 1ms
12:36:22.224 Keyboard action showNoteInfo found in database, but not in action definition.
12:36:22.224 Keyboard action showLinkMap found in database, but not in action definition.
12:36:22.225 Keyboard action toggleZenMode found in database, but not in action definition.
12:36:22.225 Keyboard action focusOnAttributes found in database, but not in action definition.
12:36:22.225 200 GET /api/keyboard-actions with 12153 bytes took 1ms
12:36:22.227 200 GET /api/script/widgets with 2885 bytes took 2ms
12:36:22.230 200 POST /api/tree/load with 2017 bytes took 1ms
12:36:22.319 200 GET /api/keyboard-shortcuts-for-notes with 2 bytes took 0ms
12:36:22.392 200 POST /api/tree/load with 5672 bytes took 1ms
12:36:22.405 200 POST /api/tree/load with 20318 bytes took 1ms
12:36:22.935 200 GET /api/note-map/_hidden/backlink-count with 11 bytes took 0ms
12:36:22.950 200 GET /api/tree?subTreeNoteId=_hidden with 27451 bytes took 1ms
12:36:24.330 200 GET /api/script/startup with 1199 bytes took 3ms
12:36:24.333 200 POST /api/tree/load with 2017 bytes took 1ms
12:36:24.347 Slow 200 PUT /api/special-notes/api-script-launcher with 318 bytes took 12ms
12:36:24.936 Slow query took 27ms: SELECT COUNT(1) FROM attributes
12:36:24.937 Table counts: notes: 4716, note_revisions: 1023, branches: 4792, attributes: 11346, etapi_tokens: 1
12:36:24.969 Slow query took 32ms: SELECT noteId as entityId FROM notes LEFT JOIN entity_changes ON entity_changes.entityName = 'notes' AND entity_changes.entityId = noteId WHERE entity_changes.id IS NULL
12:36:25.095 Slow query took 126ms: SELECT id, entityId FROM entity_changes LEFT JOIN notes ON entityId = noteId WHERE entity_changes.isErased = 0 AND entity_changes.entityName = 'notes' AND noteId IS NULL
12:36:25.127 Slow query took 25ms: SELECT noteId as entityId FROM note_contents LEFT JOIN entity_changes ON entity_changes.entityName = 'note_contents' AND entity_changes.entityId = noteId WHERE entity_changes.id IS NULL
12:36:25.189 Slow query took 32ms: SELECT branchId as entityId FROM branches LEFT JOIN entity_changes ON entity_changes.entityName = 'branches' AND entity_changes.entityId = branchId WHERE entity_changes.id IS NULL
12:36:25.224 Slow query took 35ms: SELECT id, entityId FROM entity_changes LEFT JOIN branches ON entityId = branchId WHERE entity_changes.isErased = 0 AND entity_changes.entityName = 'branches' AND branchId IS NULL
12:36:25.278 Slow query took 49ms: SELECT attributeId as entityId FROM attributes LEFT JOIN entity_changes ON entity_changes.entityName = 'attributes' AND entity_changes.entityId = attributeId WHERE entity_changes.id IS NULL
12:36:25.309 Slow query took 31ms: SELECT id, entityId FROM entity_changes LEFT JOIN attributes ON entityId = attributeId WHERE entity_changes.isErased = 0 AND entity_changes.entityName = 'attributes' AND attributeId IS NULL
12:36:26.853 Slow query took 1495ms: SELECT notes.noteId, notes.type, notes.mime FROM notes JOIN note_contents USING (noteId) WHERE isDeleted = 0 AND isProtected = 0 AND content IS NULL
12:36:26.887 All consistency checks passed with no errors detected (took 1998ms)
12:36:28.690 Sending message to all clients: {"type":"sync-push-in-progress","lastSyncedPush":179300}
12:36:28.691 Sync yvTTPC105E: Pushing 5 sync changes in 894ms
12:36:28.691 Nothing to push
12:36:28.904 Finished pull
12:36:28.905 Nothing to push
12:36:31.236 Slow query took 87ms: SELECT entityName, entityId, hash FROM entity_changes WHERE isSynced = 1 AND entityName != 'note_reordering'
12:36:31.307 Content hash computation took 158ms
12:36:31.307 Content hash checks PASSED
12:36:31.307 Sending message to all clients: {"type":"sync-finished","lastSyncedPush":179316}
12:36:58.093 304 GET /api/clipper/handshake with 45 bytes took 2ms
12:37:21.291 Nothing to push
12:37:21.467 Finished pull
12:37:21.467 Nothing to push
12:37:22.347 Slow query took 62ms: SELECT entityName, entityId, hash FROM entity_changes WHERE isSynced = 1 AND entityName != 'note_reordering'
12:37:22.430 Content hash computation took 145ms
12:37:22.430 Content hash checks PASSED
12:37:22.430 Sending message to all clients: {"type":"sync-finished","lastSyncedPush":179316}
12:37:58.103 304 GET /api/clipper/handshake with 45 bytes took 1ms
12:37:59.643 204 PUT /api/options with 0 bytes took 3ms
12:38:07.541 Slow recursive query took 21ms.
12:38:07.573 Slow 200 PUT /api/branches/none_root/expanded-subtree/0 with 133773 bytes took 53ms
12:38:07.581 200 GET /api/tree?subTreeNoteId=root with 6899 bytes took 1ms
12:38:21.298 Nothing to push
12:38:21.480 Finished pull
12:38:21.480 Nothing to push
12:38:22.498 Slow query took 82ms: SELECT entityName, entityId, hash FROM entity_changes WHERE isSynced = 1 AND entityName != 'note_reordering'
12:38:22.541 Content hash computation took 125ms
12:38:22.542 Content hash checks PASSED
12:38:22.542 Sending message to all clients: {"type":"sync-finished","lastSyncedPush":179316}
12:38:57.482 200 GET /api/tree?subTreeNoteId=_hidden with 8232 bytes took 0ms
12:38:57.700 204 PUT /api/branches/root__hidden/expanded/1 with 0 bytes took 1ms
12:38:58.110 304 GET /api/clipper/handshake with 45 bytes took 1ms
12:39:21.306 Nothing to push
12:39:21.496 Finished pull
12:39:21.497 Nothing to push
12:39:22.377 Slow query took 90ms: SELECT entityName, entityId, hash FROM entity_changes WHERE isSynced = 1 AND entityName != 'note_reordering'
12:39:22.455 Content hash computation took 168ms
12:39:22.455 Content hash checks PASSED
12:39:22.455 Sending message to all clients: {"type":"sync-finished","lastSyncedPush":179316}
12:39:58.108 304 GET /api/clipper/handshake with 45 bytes took 2ms
12:40:21.369 Nothing to push
12:40:21.572 Finished pull
12:40:21.572 Nothing to push
12:40:22.573 Slow query took 128ms: SELECT entityName, entityId, hash FROM entity_changes WHERE isSynced = 1 AND entityName != 'note_reordering'
12:40:22.630 Content hash computation took 185ms
12:40:22.630 Content hash checks PASSED
12:40:22.630 Sending message to all clients: {"type":"sync-finished","lastSyncedPush":179316}
12:40:50.879 200 GET /api/tree?subTreeNoteId=_lbRoot with 6647 bytes took 1ms
12:40:51.095 204 PUT /api/branches/_hidden__lbRoot/expanded/1 with 0 bytes took 1ms
12:40:58.105 304 GET /api/clipper/handshake with 45 bytes took 1ms
12:41:01.217 200 GET /api/tree?subTreeNoteId=_lbAvailableLaunchers with 9286 bytes took 0ms
12:41:01.431 204 PUT /api/branches/_lbRoot__lbAvailableLaunchers/expanded/1 with 0 bytes took 1ms
12:41:05.099 200 GET /api/tree?subTreeNoteId=_lbVisibleLaunchers with 20318 bytes took 1ms
12:41:05.316 204 PUT /api/branches/_lbRoot__lbVisibleLaunchers/expanded/1 with 0 bytes took 1ms
12:41:15.789 Slow 200 GET /api/search/%23workspace%20%23!template with 2 bytes took 25ms
12:41:15.819 204 PUT /api/options with 0 bytes took 3ms
12:41:21.440 Nothing to push
12:41:21.659 Finished pull
12:41:21.659 Nothing to push
12:41:22.660 Slow query took 84ms: SELECT entityName, entityId, hash FROM entity_changes WHERE isSynced = 1 AND entityName != 'note_reordering'
12:41:22.727 Content hash computation took 151ms
12:41:22.728 Content hash checks PASSED
12:41:22.728 Sending message to all clients: {"type":"sync-finished","lastSyncedPush":179316}
12:42:21.659 DB size: 274251 KB
12:42:22.758 Trusted reverse proxy: false
12:42:22.759 App HTTP server starting up at port 37840
12:42:22.759 {
  "appVersion": "0.58.7",
  "dbVersion": 212,
  "syncVersion": 29,
  "buildDate": "2023-01-17T23:14:58+01:00",
  "buildRevision": "a3149aecf41bac3c559ebbd1865e916264985ac3",
  "dataDirectory": "C:\\Users\\loren\\AppData\\Roaming\\trilium-data",
  "clipperProtocolVersion": "1.0",
  "utcDateTime": "2023-02-12T20:42:21.658Z"
}
```

**Reply by zadam** [reply]
Thanks, could reproduce the bug with collapse and fixed it.
